### ISSUE TITLE
failed to retrieve the protocol version

### ISSUE THREAD

**Issue by Fgroove** [KVM-VMI/kvm-vmi#39]
Hi, I installed the `--branch kvmi` following the [Host-based setup](https://github.com/KVM-VMI/kvm-vmi/wiki/KVM-VMI-setup#host-based-setup). 
Everything is ok until I run the test, it shows
```shell
cc@cc-Ubuntu:~/kvm-vmi/kvm/tools/kvm/kvmi$ ./kvmi-test /tmp/introspector
Waiting...
New handshake
[level=3]: failed to retrieve the protocol version (invalid authentication token?)
New handshake
[level=3]: failed to retrieve the protocol version (invalid authentication token?)
New handshake
[level=3]: failed to retrieve the protocol version (invalid authentication token?)
New handshake
[level=3]: failed to retrieve the protocol version (invalid authentication token?)
...
```
I couldn't find the reason. Any idea?

Thanks!

**Comment by Wenzel** [comment]
Hi,

Are you sure that you are running the modified kernel ?

Are you sure that `KVM_INTROSPECTION` is enabled in `/boot/config_xxx` ?

**Comment by adlazar** [comment]
@Fgroove , most probably it's a mismatch between the KVMI API used by QEMU or host kernel and kvmi-test.

Make sure (a) the proper QEMU is used to run the introspected VM (the QEMU repo has a kvmi branch too) and (b) remove the `kvmi.h` and `kvm.h` headers from `tools/kvm/kvmi/include/...` (make will re-copy the headers).

I think QEMU will fail to start (with the introspection parameters) if the kernel doesn't have KVM_INTROSPECTION.

**Comment by Fgroove** [comment]
Hi,
I checked the `/boot/config..`, and found that I mixed `KVM_INTROSPECTION` up with `KVM_INTROSPECTION_GUEST`. The former is still unset.
Then I recompiled the kernel and got the output!
![Screenshot from 2019-08-26 17-38-08](https://user-images.githubusercontent.com/27811135/63682133-ce305880-c829-11e9-9490-c97f9b6795b2.png)
It looks like working properly except `memory mapping not supported`. Is this normal?
ps. I have set `CONFIG_REMOTE_MAPPING=yes `. 

Thank you!
